app/models/user.rb

```diff
@@ -16,6 +16,10 @@ class User < ApplicationRecord
   validates :phone,
             format: { with: /\A\+1\d{10}\z/,
                       message: 'must be "+1" followed by 10 digits' }
+  before_save -> { assignments.future.destroy_all }, if: :being_deactivated?
+
+  scope :active, -> { where active: true }
```

Comment:
You mostly use active when you're querying users - why not just make this a default scope?

Reply:
I actually don't recommend default scopes. As an example, in order to get inactive users, you'd need to write `User.unscoped.inactive`, which I find clumsy and weird.

Reply:
Agreed. And if the goal is not confusing future developers, `User.inactive` always returning and empty collection is not ideal. Nor do I particularly like `User.all` returning only active users rather than, you know, "all" of them.